server/base/src/main/java/org/apache/accumulo/server/fs/FileManager.java

```diff
@@ -591,4 +593,35 @@ public synchronized int getNumOpenFiles() {
   public ScanFileManager newScanFileManager(KeyExtent tablet, CacheProvider cacheProvider) {
     return new ScanFileManager(tablet, cacheProvider);
   }
+
+  private static class FencedFile {
```

Comment:
Could TabletFile be used intead of this class?

Reply:
Good point, I can just use TabletFile. I was just trying to keep the minimal amount we needed to store in the maps but TabletFile is a pretty small object already and it would make sense to use here.